zerver/lib/push_notifications.py

```diff
@@ -680,6 +685,36 @@ def send_notifications_to_bouncer(
     )
 
 
+PUSH_NOTIFICATIONS_RECENTLY_WORKING_REDIS_KEY = "push_notifications_recently_working_ts"
+
+
+def record_push_notifications_recently_working() -> None:
+    # Record the timestamp in redisp, marking that push notifications
+    # were working as of this moment.
+    if settings.TEST_SUITE:
+        # Avoid writing to redis in tests to prevent flakiness, as tests
+        # can pollute each other's redis state.
+        return
+
+    redis_key = PUSH_NOTIFICATIONS_RECENTLY_WORKING_REDIS_KEY
+    # Keep this record around for 24h in case it's useful for debugging.
+    redis_client.set(redis_key, str(timezone_now().timestamp()), ex=60 * 60 * 24)
+
+
+def check_push_notifications_recently_working() -> bool:
+    if settings.TEST_SUITE:
+        return False
+
+    # Check in redis whether push notifications were working in the last hour.
+    redis_key = PUSH_NOTIFICATIONS_RECENTLY_WORKING_REDIS_KEY
+    timestamp = redis_client.get(redis_key)
+    if timestamp is None:
+        return False
+
+    # If the timestamp is within the last hour, we consider push notifications to be working.
+    return timezone_now().timestamp() - float(timestamp) < 60 * 60
```

Comment:
You might be able to avoid the local import if this function just lived in `remote_server.py`.